INCWaveMountainsView/Classes/INCWaveMountainsView.m

```diff
@@ -0,0 +1,591 @@
+        _shapeLayerCenterMountain = [[CAShapeLayer alloc]init];
+        
+        [_shapeLayerCenterMountain setFillColor:[UIColor clearColor].CGColor];
+        [_shapeLayerCenterMountain addSublayer:self.gradientCenterMountain];
+
+        [self.layer addSublayer:_shapeLayerCenterMountain];
+    }
+    return _shapeLayerCenterMountain;
+}
+
+-(CAShapeLayer *)shapeLayerRightMountain
+{
+    if (!_shapeLayerRightMountain) {
+        _shapeLayerRightMountain = [[CAShapeLayer alloc]init];
+        
+        [_shapeLayerRightMountain setFillColor:[UIColor clearColor].CGColor];
+        [_shapeLayerRightMountain addSublayer:self.gradientRightMountain];
+
+        [self.layer addSublayer:_shapeLayerRightMountain];
+    }
+    return _shapeLayerRightMountain;
+}
+
+-(NSMutableDictionary *)percentReachFull
+{
+    if (!_percentReachFull) {
+        _percentReachFull = [[NSMutableDictionary alloc]init];
+    }
+    return _percentReachFull;
+}
+
+-(void)_removeThePointWithIdPoint:(NSInteger)idPoint inMountain:(INCMountainPosition)mountain
+{
+    //In order to reset the values we need to reset the values to 0 and then remove the point.
+    
+    __weak typeof(self) weakSelf = self;
+    dispatch_group_t group = dispatch_group_create();
+    
+    dispatch_group_enter(group);
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        UIBezierPath *path = [self _pathWithIdPoint:idPoint percent:0 inMountain:mountain];
+        if (path) {
+            [self _animatePath:path inMountain:mountain];
+        }
+        dispatch_group_leave(group);
+    });
+    
+    dispatch_group_notify(group, dispatch_get_main_queue(), ^{
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+
+            [weakSelf _removeMountainPosition:mountain];
+        });
+    });
+}
+
+-(void)_removeMountainPosition:(INCMountainPosition)mountainPosition
+{
+    switch (mountainPosition) {
+        case INCMountainPositionLeft:
+            if (self.leftMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:self.leftMountainPointId];
+            }
+            self.leftMountainPercent = -1;
+            self.leftMountainPointId = NULL;
+            break;
+        
+        case INCMountainPositionCenter:
+            if (self.centerMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:@(self.centerMountainPercent)];
+            }
+            self.centerMountainPercent = -1;
+            self.centerMountainPointId = NULL;
+            break;
+            
+        case INCMountainPositionRight:
+            if (self.rightMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:@(self.rightMountainPercent)];
+            }
+            self.rightMountainPercent = -1;
+            self.rightMountainPointId = NULL;
+            
+            break;
+    }
```

Comment:
Instead of having a switch where you repeat the code, you could have a dictioniary where you would return the percentages